parallel-consumer-core/src/main/java/io/confluent/parallelconsumer/internal/ConsumerManager.java

```diff
@@ -34,6 +37,8 @@ public class ConsumerManager<K, V> {
      */
     private ConsumerGroupMetadata metaCache;
 
+    private final AtomicReference<Set<TopicPartition>> pausedPartitionsCache = new AtomicReference<>();
```

Comment:
I feel like we can just keep track of the size itself.
```suggestion
    @Getter
    private final AtomicInteger pausedPartitionSize = new AtomicInteger(0);
```

Reply:
I reflected the review except for few changes.

- Use `volatile int` type and named `pausedPartitionSizeCache` for consistency
- Add getter named `get PausedPartitionSize()` returning `pausedPartitionSizeCache` value